### PR TITLE
feat: Start-Stop providers based on the provider market config

### PR DESCRIPTION
**Pull request by davidterpay**
closes BLO-956

### CODE REVIEW

File: oracle/orchestrator/lifecycle.go [hunk 1]
```diff
@@ -33,73 +24,67 @@ func (o *ProviderOrchestrator) Start(ctx context.Context) error {
...
-
-		o.logger.Info("provider orchestrator exited successfully")
 	}
 
-	return nil
+	o.wg.Wait()
+	o.logger.Info("provider orchestrator exited successfully")
 }
 
 // execProviderFn returns a function that starts the provider. This function is used
```
Comment (aljo242) [inline oracle/orchestrator/lifecycle.go]: nit: function no longer returns a function

File: oracle/orchestrator/market_mapper.go [hunk 2]
```diff
@@ -11,53 +11,52 @@ import (
...
-	return func() error {
-		mapper := o.GetMarketMapProvider()
-		ids := mapper.GetIDs()
-		if len(ids) != 1 {
-			o.logger.Error("market map provider can only be responsible for one chain", zap.Any("ids", ids))
-			return nil
-		}
+func (o *ProviderOrchestrator) listenForMarketMapUpdates(ctx context.Context) {
+	mapper := o.GetMarketMapProvider()
+	ids := mapper.GetIDs()
```
Comment (aljo242) [inline oracle/orchestrator/market_mapper.go]: mmProvider

File: providers/base/provider.go [hunk 3]
```diff
@@ -100,12 +100,14 @@ func NewProvider[K providertypes.ResponseKey, V providertypes.ResponseValue](opt
 // Start starts the provider's main loop. The provider will fetch the data from the handler
 // and continuously update the data. This blocks until the provider is stopped.
 func (p *Provider[K, V]) Start(ctx context.Context) error {
-	p.logger.Info("starting provider")
+	if ctx == nil {
+		p.logger.Info("context is nil; exiting")
+		return nil
```
Comment (aljo242) [inline providers/base/provider.go:105]: shouldn't this be an error?

File: oracle/oracle.go [hunk 4]
```diff
@@ -181,6 +178,15 @@ func (o *OracleImpl) fetchPrices(provider types.PriceProviderI) {
 		}
 	}()
 
+	if !provider.IsRunning() {
```
Comment (Eric-Warehime) [inline oracle/oracle.go:181]: This just reads kinda weird. If the provider wasn't running wouldn't fetchPrices not be called?

Comment (Eric-Warehime) [reply oracle/oracle.go:181]: Maybe what we want here is some kind of context check or something?

File: oracle/orchestrator/lifecycle.go [hunk 5]
```diff
@@ -33,73 +24,67 @@ func (o *ProviderOrchestrator) Start(ctx context.Context) error {
...
-			o.logger.Error("provider orchestrator exited with error", zap.Error(err))
-			return err
-		}
-
-		o.logger.Info("provider orchestrator exited successfully")
 	}
 
-	return nil
+	o.wg.Wait()
+	o.logger.Info("provider orchestrator exited successfully")
```
Comment (Eric-Warehime) [inline oracle/orchestrator/lifecycle.go:66]: Will the logger provide context here such as which provider this is, or will the log line just be "provider exited"? If not I think we should add some context.

**Review by aljo242 — APPROVED** [review]

**Pull request merged**